expenseease-group-svc/src/main/java/com/expenseease/group/service/GroupService.java

```diff
@@ -66,9 +68,29 @@ public Set<GroupDTO> findGroupsForUser(Long userId) {
     public Set<UserDTO> findUsersInGroup(Long groupId) {
         Group group = groupRepository.findById(groupId).get();
         if (group == null) {
-            // Send Bad Response
+            throw new InvalidGroupException("Group doesn't exist with id: "+groupId);
         }
         return group.getUsers().stream()
                 .map(u -> new UserDTO(u.getId(), u.getName(), u.getEmail())).collect(Collectors.toSet());
     }
+
+    public void deleteGroup(Long groupId) {
+        Group group = groupRepository.findById(groupId)
+                .orElseThrow(InvalidGroupException::new);
+        groupRepository.delete(group);
+    }
+
+    public GroupDTO removeUserFromGroup(Long userId, Long groupId){
+        Group group = groupRepository.findById(groupId).orElse(null);
+        if (group == null) {
+            throw new InvalidGroupException("Group doesn't exist with id: "+groupId);
+        }
+        User user = group.getUsers().stream().filter(users -> users.getId() == userId).findFirst().orElse(null);
+        if(user == null){
+            throw new InvalidUserException("User not exist for id: "+userId);
+        }
+        group.getUsers().remove(user);
+        groupRepository.save(group);
+        return Utility.mapObject(group, GroupDTO.class);
```

Comment:
Is this tested?
we can not use this mapper, as our model has dependencies.
Group depends on User and User depends on Group. There will be infinite mapping.
We need create the dto by iterating the model obj.